### ISSUE TITLE
Issue while generating Siebel image

### ISSUE THREAD

**Issue by nachosandoval** [OracleSiebel/ConfiguringSiebel#147]
Hello,

While I'm generating the Siebel image running build-mde-multistage, I'm getting this error:
The command '/bin/sh -c bash /config/getMDE.sh installers 22.12' returned a non-zero code: 8

Debugging the wget from the installers container shows that the Siebel installer files are being downloaded.

Any idea why?

Thanks!

**Comment by OracleSiebel** [comment]
Can you replicate the result of 8 by issuing the wget command directly and then immediately echoing the result? To see the result, use:


`
echo $?
`

You say, it's downloading something, but have you verified it's get everything? Could it be a disk space issues?

**Comment by OracleSiebel** [comment]
I just tested with the very latest nginx container and, for me, the download of the installer works fine and it moves directly on to the installation step.


```
Downloading installer
+ wget -r -q -np -nH --cut-dirs=2 -l 0 -R 'index.html*' http://installers/siebel/22.9/Siebel_Enterprise_Server/
Wed Feb  8 18:05:41 UTC 2023
Installing Siebel ...
```

I'll double check with 22.12 just in case there's some oddity with that installer.

When you next test, perhaps you could attach the entire build-all-multistage log file. It seems clear you are successfully downloading the DB client and installing that, but it would be good to confirm.

**Comment by nachosandoval** [comment]
Thanks for your quick reply!

I first tried with 23.1 and then moved to 22.12 just to see if there was some oddity with the latest version, but the behaviour seems to be the same.

Indeed it's downloading the client, and that part works fine.

When I set the wget to -v, I get the following info:

2023-02-09 07:44:47 (281 MB/s) - 'stage/Components/oracle.swd.oui/12.2.0.1.4/1/DataFiles/Expanded/oui/instImages/oneclickbg.gif' saved [15305/15305]

--2023-02-09 07:44:47--  http://installers/siebel/Disk1/stage/Components/oracle.swd.oui/12.2.0.1.4/1/DataFiles/Expanded/oui/instImages/splashScr.gif
Reusing existing connection to installers:80.
HTTP request sent, awaiting response... 200 OK
Length: 30502 (30K) [image/gif]
Saving to: 'stage/Components/oracle.swd.oui/12.2.0.1.4/1/DataFiles/Expanded/oui/instImages/splashScr.gif'

     0K .......... .......... .........                       100%  342M=0s

2023-02-09 07:44:47 (342 MB/s) - 'stage/Components/oracle.swd.oui/12.2.0.1.4/1/DataFiles/Expanded/oui/instImages/splashScr.gif' saved [30502/30502]

FINISHED --2023-02-09 07:44:47--
Total wall clock time: 1m 55s
Downloaded: 7033 files, 10G in 1m 50s (93.5 MB/s)
The command '/bin/sh -c mkdir -p -ma+rwx /siebel/ses /siebel/oraInventory  /stage && groupadd -g  1000 siebel && useradd -m -g siebel -G siebel -u 1000 siebel && cd /mnt && date && echo Downloading installer && bash /config/getMDE.sh ${INSTALLSITE} ${VERSION}&& chmod a+rx /config && chown -R siebel:siebel /config && date && chmod +x /mnt/Siebel_Enterprise_Server/Disk1/install/unzip && chmod +x /mnt/Siebel_Enterprise_Server/Disk1/install/runInstaller && cd / && echo Installing Siebel ... && bash /config/mde_install.sh && rm -rf /mnt/Siebel_Enterprise_Server/ && echo Waiting for deployment of web archives ... && sleep 15 && chown -R siebel:siebel /siebel && chmod -R 755 /siebel/ses && sed -i 's/-t 90/-t 30/g' /siebel/mde/siebsrvr/admin/common.scm && sed -i 's/-t 90/-t 30/g' /siebel/mde/gtwysrvr/admin/common.scm && cd /siebel/mde/applicationcontainer_external/siebelwebroot && mkdir -p ara chs cht csy dan deu enu esn fin fra heb ita jpn kor nld plk ptb ptg rus sve tha trk && bash /config/processRemovals /config/siebelRemovalList' returned a non-zero code: 8

And there seems to be enough space:

ubuntu@ip-172-31-24-42:~$ df -h
Filesystem      Size  Used Avail Use% Mounted on
/dev/root       146G   84G   62G  58% /
[build.2023-02-09_07_39.zip](https://github.com/OracleSiebel/ConfiguringSiebel/files/10694861/build.2023-02-09_07_39.zip)

**Comment by OracleSiebel** [comment]
You've made more changes than you disclosed initially. Clearly, you've modified getMDE.sh to match your setup. This is all fine, and expected, but those changes mean you now have to own your changes.

Looking at your log file I see your wget command looks like this:

`wget -v -r -np -nH --cut-dirs=2 -l 0 http://installers/siebel/`

If you were running the supplied code, it would look like this (from my log file):

`wget -r -q -np -nH --cut-dirs=2 -l 0 -R 'index.html*' http://installers/siebel/22.12/Siebel_Enterprise_Server/`

Let's try to isolate this from the long winded build system and focus instead on the wget command. I have the 'installers' container running locally. Therefore to test the wget command in isolation I ran this:

`wget -v -r -np -nH --cut-dirs=2 -l 0 -R 'index.html*' http://localhost/siebel/22.12/Siebel_Enterprise_Server/ > wget.txt 2>&1`

Immediately afterward, execute this:

`echo $?`

For me, the return value is 0. For you, I expect the return value to be 8. Looking through your log file I see this:

```
2023-02-09 07:42:53--  http://installers/siebel/STRING_LINK_TO_UPDATE_GUIDES
Reusing existing connection to installers:80.
HTTP request sent, awaiting response... 404 Not Found
2023-02-09 07:42:53 ERROR 404: Not Found.
```

I suspect this is the actual cause of the error. I have a vague recollection of seeing this in the past, but I can't remember why it happened. Looking at my nginx.conf I see it does differ a little from the one here on GitHub. I have this, and I see I have something relating to symlinks there. Give this a go:

```
worker_processes 1;
error_log nginx_error.log;
events {
    worker_connections 1024;
}

http {
  include /etc/nginx/mime.types;
  disable_symlinks off;
  server {
        listen   80;
        server_name  local.com;
        root   /usr/share/nginx/html;
        autoindex on;
        location / {
               autoindex on;
        }
  }
} 

```
For me, I simply downloaded 22.12 from support.oracle.com and unpacked the contents in a new 22.12/Siebel_Enterprise_Server folder and ran the build.
